src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -223,11 +223,15 @@ private async Task AddProjectItemAsync(string filePath)
             container.AddFromFileCopy(fullPath);
         }
 
-        public override async Task<bool> UninstallPackageAsync(
+        public override Task<bool> UninstallPackageAsync(
             PackageIdentity packageIdentity, INuGetProjectContext _, CancellationToken token)
         {
-            await ProjectServices.References.RemovePackageReferenceAsync(packageIdentity.Id);
+            return UninstallPackageAsync(packageIdentity.Id);
```

Comment:
I'm not sure how this works, but if the user's intent is to uninstall the package for a specific target framework, then it seems this method would uninstall the package from all target frameworks that depend on this package.

Reply:
This is legacy anyways, which does not support mutlitargeting.